fvm/fvm_blockcontext_test.go

```diff
@@ -1033,13 +1033,30 @@ func TestBlockContext_ExecuteTransaction_InteractionLimitReached(t *testing.T) {
 				accounts, err := testutil.CreateAccounts(vm, view, derivedBlockData, privateKeys, chain)
 				require.NoError(t, err)
 
-				txBody := testutil.CreateContractDeploymentTransaction(
+				// fund account so the payer can pay for the next transaction.
+				txBody := transferTokensTx(chain).
+					SetProposalKey(chain.ServiceAddress(), 0, 0).
+					AddAuthorizer(chain.ServiceAddress()).
+					AddArgument(jsoncdc.MustEncode(cadence.UFix64(100_000_000))).
+					AddArgument(jsoncdc.MustEncode(cadence.NewAddress(accounts[0]))).
+					SetPayer(chain.ServiceAddress())
+
+				err = testutil.SignEnvelope(txBody, chain.ServiceAddress(), unittest.ServiceAccountPrivateKey)
+				require.NoError(t, err)
+				tx := fvm.Transaction(txBody, derivedBlockData.NextTxIndexForTestingOnly())
+				err = vm.Run(ctx, tx, view)
+				require.NoError(t, err)
+				assert.NoError(t, tx.Err)
+
+				ctx.MaxStateInteractionSize = 500_000
+
+				txBody = testutil.CreateContractDeploymentTransaction(
 					"Container",
 					script,
 					accounts[0],
 					chain)
 
-				txBody.SetProposalKey(chain.ServiceAddress(), 0, 0)
+				txBody.SetProposalKey(chain.ServiceAddress(), 0, 1)
```

Comment:
instead of using magic number, maybe create a seqNum variable to track the number